### ISSUE TITLE
A Survey on GANs for Anomaly Detection

### ISSUE THREAD

**Issue by llien30** [mn1204/paper_summary#7]
# INFO
A survey about [AnoGAN](https://github.com/mn1204/paper_summary/issues/4#issue-558855326), [EGBAD](https://github.com/mn1204/paper_summary/issues/3#issue-558830198), [GANomaly](https://github.com/mn1204/paper_summary/issues/1#issue-558663101)
(The link is connected to my paper summary)
## Author
Federico Di Mattia, Paolo Galeone, Michele De Simoni, Emanuele Ghelf
## Conference or Year
Submitted on 27 Jun 2019
## Link
- [arxiv](https://arxiv.org/abs/1906.11632)
- [official GitHub]()

# Abstract
- make the empirical validation of the main GAN models for anomaly detection
- increase the experimental results on different datasets
- public the complete Open Source toolbox for Anomaly Detection using GANs

# Introduction
the task of **Anomaly Detection** is ...
_the problem of finding patterns in data that do not conform to expected behavior_([Chandola et al.](http://cinslab.com/wp-content/uploads/2019/03/xiaorong.pdf))
- **GANs**
GANs can model complex and high dimensional real-world data
→GANs can be used to anomaly detection (but the application is only recently explored…)
all the GAN-based approach to Anomaly Detection is based on the Adversarial Feature Learning idea (proposed when BiGAN is published)
**BiGAN = GANs + inverse mapping**

### GANs
- Original GANs([Goodfellow et al., 2014](https://arxiv.org/abs/1406.2661))
the min-max game of the Generator and the Discriminator
<img width="694" alt="スクリーンショット 2020-02-05 08 21 48" src="https://user-images.githubusercontent.com/51707667/73796447-8f4af980-47f0-11ea-88aa-141676585313.png">

- Conditional GANs([Mirza & Osindero, 2014](https://arxiv.org/abs/1411.1784))
<img width="710" alt="スクリーンショット 2020-02-05 08 15 00" src="https://user-images.githubusercontent.com/51707667/73796082-9f160e00-47ef-11ea-918c-ad894811abf7.png">

- BiGAN([Donahue et al., 2016](https://arxiv.org/abs/1605.09782))
this network have Encoder(E)
trains mapping both **latent space → data**, **data → latent space** simultaneously
<img width="717" alt="スクリーンショット 2020-02-05 08 20 36" src="https://user-images.githubusercontent.com/51707667/73796391-64f93c00-47f0-11ea-8f4f-8d4d1aa914fc.png">
<img width="912" alt="スクリーンショット 2020-02-05 08 19 35" src="https://user-images.githubusercontent.com/51707667/73796326-42ffb980-47f0-11ea-9108-e2af1054376f.png">

## GANs for Anomaly Detection
<img width="754" alt="スクリーンショット 2020-02-05 08 18 12" src="https://user-images.githubusercontent.com/51707667/73796225-1055c100-47f0-11ea-8985-5eb9806d3eae.png">

### AnoGAN
**Anomaly Score**
<img width="711" alt="スクリーンショット 2020-02-05 08 18 46" src="https://user-images.githubusercontent.com/51707667/73796265-22cffa80-47f0-11ea-98af-ac0a09204d59.png">

 **pros**
<img width="717" alt="スクリーンショット 2020-02-05 08 23 28" src="https://user-images.githubusercontent.com/51707667/73796541-cae5c380-47f0-11ea-8162-a8c217bb25e9.png">

**cons**
<img width="704" alt="スクリーンショット 2020-02-05 08 24 44" src="https://user-images.githubusercontent.com/51707667/73796600-fa94cb80-47f0-11ea-8196-374a7a3466fd.png">

<img width="809" alt="スクリーンショット 2020-02-05 08 32 44" src="https://user-images.githubusercontent.com/51707667/73796972-164ca180-47f2-11ea-9ab7-082b47046316.png">

### Efficient GAN(EGBAD)
tried to solve the AnoGAN disadvantages with BiGAN
**pros**
- can computing the anomaly score without optimization steps during the inference
→ AnoGAN need iteration

### GANomaly
**Loss Function**
<img width="683" alt="スクリーンショット 2020-02-05 08 40 49" src="https://user-images.githubusercontent.com/51707667/73797583-d090d880-47f3-11ea-8997-429d02a4bcd2.png">

**pros**
- An encoder is learned during the training process → no optimization steps like AnoGAN
- Using an auto-encoder like architecture (no use of noise prior) makes the entire learning process faster
- The anomaly score is easier to interpret
- The contextual loss can be used to localize the anomaly

**cons**
- can detect anomalies both in the image space and in the latent space
     → but the anomaly score is more higher to calculate only in the latent space
     → can be associated with a generated sample with a low contextual loss value and thus very similar to the input - and vice versa(?)
- Defines a new anomaly score
<img width="931" alt="スクリーンショット 2020-02-05 08 59 24" src="https://user-images.githubusercontent.com/51707667/73798325-d12a6e80-47f5-11ea-958c-7f1df57bd437.png">

# Experiments
re-implemented all of the three network to evaluate(with Tensorflow)
### Dataset
<img width="736" alt="スクリーンショット 2020-02-05 10 21 50" src="https://user-images.githubusercontent.com/51707667/73802601-7ac32d00-4801-11ea-9c11-6a5c544396d3.png">

### Models
- EGBAD
      image size : 28 x 28
- GANomaly
      image size : resized to 32 x 32
- AnoGAN
      testing this model is time consuming because it needs a lot of times iteration
      → no test data in this paper(should refer original AnoGAN paper)
 
## Result
<img width="872" alt="スクリーンショット 2020-02-05 10 25 54" src="https://user-images.githubusercontent.com/51707667/73802751-ec9b7680-4801-11ea-94b1-1d3907293906.png">

# Conclusion
- verified the effectiveness of the GANs-based approach to the anomaly detection problem
- show the differences between the original papers and the publicly available code

## Date
2020/02/05
